### ISSUE TITLE
DRM

### ISSUE THREAD

**Issue by christchron** [micnews/react-jw-player#79]
can we use DRM on your library?

**Comment by danmakenoise** [comment]
Yeah, you should be able to include DRM in your playlist object! See [here](https://support.jwplayer.com/customer/portal/articles/2561182)

**Comment by christchron** [comment]
then we can't use just file = xxx.m3u8?
we need to use playlist and pass the file there?

**Comment by danmakenoise** [comment]
@christchron right now, I believe that is the only way. Take a look at those docs I linked to and tell me which DRM method you are planning on using. If you show me what the JW Player code would be, I can make a PR to this library to support it.

**Comment by christchron** [comment]
I need to use widewive for now. But later maybe need something else for Apple/Safari.

`playlist: [{
    sources: [{
      file: '//www.website.com/media/videofile.mpd',
        drm: {
          widevine: {
          url: '//widevine-proxy.appspot.com/proxy',
            headers:[{
              name:”customData”,
              value  : "abcdefg1234567hijklmn89101112opqrs98765tuvwxyz"
            }]
          }
        }
    },{`

If possible, i need to use only file but with drm. @danmakenoise .

Thanks btw

**Comment by christchron** [comment]
Hello @danmakenoise, can you help me?

**Comment by danmakenoise** [comment]
@christchron hey! I don't have any experience with DRM and JW Player! It does seem like we may need to add some new functionality to `react-jw-player` to accommodate this, though.

What I need from you is to use JW Player's docs and find a way to handle the DRM _without_ `react-jw-player`, using the normal JWPlayer SDK. If you can do that, and show the code here, then I can help get this functionality folded into this library.

**Comment by vishwajeetv** [comment]
@danmakenoise @randichilyon  it is rather very simple.
You can pass appropriate DRM configuration in `drm` props
like this

```
<ReactJWPlayer 
playerScript={'<player-script-link>'}
autostart='viewable'
playlist={<playlist>}
playerId = {'abc'}
drm = {drm}  />
```
Where 
`drm = {clearKey: {key: '<key>', keyId: '<keyId>'}}`
in the case of clearKey DRM

**Comment by zunairzubair3** [comment]
@danmakenoise Any update on this?

**Comment by SarfarazAhmed1012** [comment]
any updates?
